### ISSUE TITLE
Can't enable all rules

### ISSUE THREAD

**Issue by SamuelCabralCruz** [mgechev/revive#535]
**Describe the bug**

Can't enable all rules at once

**To Reproduce**
1. I updated revive `go get -u github.com/mgechev/revive`
2. I run it with the following flags & configuration file:

```shell
revive -config revive.toml -formatter friendly ./...
```

```toml
ignoreGeneratedHeader = false
severity = "error"
confidence = 0.8
errorCode = 1
warningCode = 0

enableAllRules = true
```

**Expected behavior**

All rules are being enabled and used to lint my project without error.

**Logs**

![image](https://user-images.githubusercontent.com/15621634/123643014-f4dac400-d7f1-11eb-82b7-d1bf61f3e3d9.png)

**Desktop (please complete the following information):**
 - OS: 
![image](https://user-images.githubusercontent.com/15621634/123643314-408d6d80-d7f2-11eb-9215-38d733c732c6.png)
 - Version of Go: 1.16

**Additional context**

**Comment by chavacava** [comment]
Hi @SamuelCabralCruz, thanks for filling the issue.

As described in the documentation (see README.md at section Configuration)

>When enabling all rules you still need/can provide specific configurations for rules. 

In your case, `revive` is telling you that you need to provide a configuration for the rule `argument-limit`. Why? because the that rule (and a few others) needs configuration to be executed. In the case of `argument-limit`you need to indicate the max number of argument that is tolerated for function definitions.

The documentation provides an example of a configuration like the one you are trying to setup:

```toml
severity = "warning"
confidence = 0.8
errorCode = 0
warningCode = 0

# Enable all available rules
enableAllRules = true

# Disabled rules
[rule.blank-imports]
    Disabled = true
[rule.file-header]
    Disabled = true
[rule.max-public-structs]
    Disabled = true
[rule.line-length-limit]
    Disabled = true
[rule.function-length]
    Disabled = true

# Rule tunning
[rule.argument-limit]
    Arguments = [5]
[rule.cyclomatic]
    Arguments = [10]
[rule.cognitive-complexity]
    Arguments = [7]
[rule.function-result-limit]
    Arguments = [3]
```

Please let me know if you are able to resolve your problem by following the above indications.

**Comment by SamuelCabralCruz** [comment]
@chavacava ok yeah I understand now! I tried the proposed solution and it did work as expected. I still think that all rules should have a prescribed value which would allow to use them without additional config. This would also be more use friendly and less confusing for new comers like me! 

The following should be enough to benefit from all the goodies of revive imho.

```
severity = "warning"
confidence = 0.8
errorCode = 0
warningCode = 0
enableAllRules = true
```

Is there a specific reason why there are some rules that needs configuration?

**Comment by chavacava** [comment]
Great!

>  I still think that all rules should have a prescribed value which would allow to use them without additional config

This is a topic we have already discussed and we opted to do not set _default_ values for configurations because these defaults will certainly do not match with the style rules defined in the majority of code bases out there. For example, which value should we set as a default for `argument-limit`? 3, 4, 10, 12 ? And what about `cyclomatic` complexity, or `line-length-limit`? All these are very opinionated subjects

> Is there a specific reason why there are some rules that needs configuration?
As explained before, these rules do not have default limits, thus to run, they do need configuration setting those limits.

> This would also be more use friendly and less confusing for new comers like me!
We try to make `revive` as friendly as possible (`enableAllRules` option was added for that). We also put effort in the documentation to describe as clear as possible how to take advantage of all the functionalities of the tool.

**Comment by SamuelCabralCruz** [comment]
@chavacava thanks for the quick feedback. it is really appreciated. I do agree that these subjects are really opinionated, but having the possibility to override the default value if it does not fit my needs is good enough to me! But never the less I really enjoy using revive. Thanks for your time and the great work 💪
